simple-stack-example-kotlin-masterdetail-fragments/src/main/java/com/example/ktdagger/userdetail/UsersFragment.kt

```diff
@@ -0,0 +1,132 @@
+import kotlinx.android.synthetic.main.view_loading.loadingView
+import javax.inject.Inject
+
+class UsersFragment : BaseFragment() {
+
+    @Inject
+    lateinit var apiService: ApiService
+
+    @Inject
+    lateinit var compositeDisposable: CompositeDisposable
+
+    @Inject
+    lateinit var schedulerProvider: BaseSchedulerProvider
+
+    private lateinit var realm: Realm;
+    private lateinit var autoCompleteAdapter: UsersAdapter
+    private var indexOfContentView: Int = 0
+    private var indexOfLoadingView: Int = 0
+    private lateinit var items: RealmResults<UserRO>
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_users, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        AndroidApp.appComponent?.inject(this);
+
+        indexOfContentView = viewAnimator.indexOfChild(contentView)
+        indexOfLoadingView = viewAnimator.indexOfChild(loadingView)
+
+        realm = Realm.getDefaultInstance()
+        val items = realm.where<UserRO>().sort("userName", Sort.ASCENDING).findAll()
+        autoCompleteAdapter = UsersAdapter(realm, items)
+        usersSearchView.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(p0: Editable?) {
+                autoCompleteAdapter.updateInput(p0.toString())
+            }
+
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+        });
+        contactsView.adapter = autoCompleteAdapter
+        contactsView.layoutManager = LinearLayoutManager(view.context, VERTICAL, false)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        loadUsers()
+    }
+
+    private fun loadUsers() {
+        loadingView.visibility = View.VISIBLE
+        compositeDisposable += apiService
+            .getAllUsers()
+            .retry()
+            .subscribeOn(schedulerProvider.io())
+            .doOnNext { allUserResponse: AllUsersResponse ->
+                println("Thread doOnNext: %s" + Thread.currentThread().name)
+                val userROs: MutableList<UserRO> = ArrayList();
+                for (user in allUserResponse.result) {
+                    val userRO: UserRO = UserRO.createFromUser(user);
+                    userROs.add(userRO);
+                }
+                Realm.getDefaultInstance().use { realmInstance ->
+                    realmInstance.executeTransaction {
+                        val result: UserRO? = realmInstance.where<UserRO>().findFirst()
+                        if (result == null) {
+                            realmInstance.createObject<UserRO>(0)
```

Comment:
Is that always a valid primary key value?

Reply:
in this example, 1 would be since it is incremental from 1. I will change that.